v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/azure-service-operator/v2/tools/asoctl/internal/client"
+)
+
+func CleanDeprecatedCRDVersions(ctx context.Context) error {
+	cl, err := client.NewClient()
+	if err != nil {
+		return err
+	}
+
+	list, err := cl.CustomResourceDefinitions().List(ctx, v1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	if list.Items == nil || len(list.Items) == 0 {
+		fmt.Println("found 0 results, make sure you have ASO CRDs installed")
+	}
+
+	var updated bool
+	for _, item := range list.Items {
+		crdName := item.Name
+		crdRegexp := regexp.MustCompile(".*\\.azure\\.com")
+		deprecatedVersionRegexp := regexp.MustCompile("v1alpha1api.{8}.*")
```

Comment:
@matthchr and I had a conversation on this already. We don't want the end user to specify what they want to delete/deprecate(which can break their cluster). The tool we're providing is just to remove `storedVersion` for the deprecated CRDs which we specify in the tool. 

For sure we want a workaround to solve the problem dynamically where in the future we'll be deprecating `v1beta` which'll be a few versions after we go GA(I assume).

Reply:
The logic that got me to "we just shouldn't accept users parameters here (at least initially)" is:

1. This is a dangerous operation because we're removing stored versions which is possibly destructive, so we're going to need a lot of validation on what the user passes in to make sure they don't remove _all_ of their storage versions (or really any storage version that is "in use")
2. The operator pod expects a certain set of versions to exist based on its version (`alpha.6` expects all of alpha versions, `beta.0` expects all of the beta versions but doesn't look for the alpha versions (it only cares about stored version), and `stable` will expect the stable versions and presumably not care about beta or alpha). If the expected versions are missing, the pod will crash/not function.

Achieving those two validations means that the tool is going to need to introspect the cluster to determine exactly what versions can be safely removed. At that point, it seemed pointless to give the user any real control because we'd just be making them guess at what we allowed, and each currently existing version of ASO has exactly one right "allowed set" of removals. We could just take no input and remove exactly what is supposed to be removed.

I do agree with @theunrepentantgeek's sentiment that we should structure the code such that it can be expanded in the future to possibly support the `--remove` syntax without needing a huge internal refactor, but I think exposing such an option now when the operator pod expects certain versions of certain CRDs is a bad idea because it would be very easy to remove something you shouldn't and put the pod into crashloop backoff.

OTOH: `--remove` can already be done via `kubectl` assuming that the user wants to remove the entire CRD - the unique issue that this tool is solving is removing just one version of a CRD rather than the whole CRD. I don't think that necessarily means we don't need to to worry about structuring things to be expandable (we always should just do that by default), but I am not sure that the `--remove` is going to be needed for whole CRDs because that already exists in `kubectl`. It's specifically versions within a CRD that needs this tool.